# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1466407270 18000
#      Mon Jun 20 02:21:10 2016 -0500
# Node ID 6be06f0c7463ce0f690ac9820f72142446a4e942
# Parent  d5a58720459d1c34fd0a3aefa26631ba8ae7cc5f
gnuplot graphics: Fix MarkerFaceColor invalid and MarkerEdgeColor 'none' bugs (bug #47974).

* __gnuplot_draw_axes__.m: If after calling do_linestyle_command() the style
  is empty, use 'points' as a placeholder point-type but leave the data empty.
  (gnuplot_pointtype): Treat '.' point type as having no marker face and
  therefore no color.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -627,16 +627,20 @@ function __gnuplot_draw_axes__ (h, plot_
         [style, sidx] = do_linestyle_command (obj, obj.color, data_idx,
                                               plot_stream, errbars);
         if (__gnuplot_has_feature__ ("linetype"))
           scmd = "linetype";
         else
           scmd = "linestyle";
         endif
 
+        if isempty (style{1})
+          style{1} = "points";
+          data{data_idx} = {};
+        endif
         withclause{data_idx} = sprintf ("with %s %s %d",
                                         style{1}, scmd, sidx(1));
 
         if (length (style) > 1)
           data_idx += 1;
           is_image_data(data_idx) = is_image_data(data_idx - 1);
           parametric(data_idx) = parametric(data_idx - 1);
           have_cdata(data_idx) = have_cdata(data_idx - 1);
@@ -1308,17 +1312,22 @@ function __gnuplot_draw_axes__ (h, plot_
                 have_cdata(data_idx) = true;
               else
                 have_cdata(data_idx) = false;
               end
               have_3d_patch(data_idx) = false;
               titlespec{data_idx} = tspec;
               usingclause{data_idx} = sprintf ("record=%dx1 using ($1):($2):($3)%s",
                                                yrec, ccol);
-              data{data_idx} = zz;
+              if isempty (style{i_stl})
+                style{i_stl} = "points";
+                data{data_idx} = {};
+              else
+                data{data_idx} = zz;
+              endif
               withclause{data_idx} = sprintf ("with %s %s %s",
                                               style{i_stl}, scmd, sopt);
             endfor
           endfor
         endif
 
       case "text"
         do_text (plot_stream, gnuplot_term, enhanced, obj, h);
@@ -2007,22 +2016,17 @@ function [pt, pt2, obj] = gnuplot_pointt
       case "+"
         pt = pt2 = "1";
       case "o"
         pt = "6";
         pt2 = "7";
       case "*"
         pt = pt2 = "3";
       case "."
-        pt = "6";
-        pt2 = "7";
-        if (isfield (obj, "markerfacecolor")
-            || strcmp (obj.markerfacecolor, "none"))
-          obj.markerfacecolor = "auto";
-        endif
+        pt = pt2 = "7";
         if (isfield (obj, "markersize"))
           obj.markersize /= 3;
         else
           obj.markersize = 5;
         endif
       case "x"
         pt = pt2 = "2";
       case {"square", "s"}
